specs/plugin-extensibility.md

```diff
@@ -22,13 +22,28 @@ Keys and associated certificates used for signing artifacts using Notation could
 
 Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads. This mechanism is used as Go language (used to develop [Notation Go library](https://github.com/notaryproject/notation-go)) does not have a [in built support](https://github.com/golang/go/issues/19282) to load and execute plugins that works across OS platforms. Other mechanisms like gRPC require every plugin to be implemented as a service/daemon.
 
+### Package and distribute plugin file
```

Comment:
It's better to give an overview about what is a plugin file, the recommendation for distribution with examples, and then some specific conventions. For example,

A Notation plugin file is an executable file with the naming convention `notation-{plugin-name}`. The plugin file can be distributed as a single executable file. However, it is highly recommended to package the plugin file in an archive file with compression. Plugin vendor may add other files in the same compressed archive file. The following is an example of the layout of the compressed archive file:

```
- notation-{plugin-name}.tar.gz
   - notation-{plugin-name}
   - LICENSE 
```

Then the detailed conventions (My draft language)
- On Linux, the plugin file should be executable with file permission `rx`. On Windows, it should be `.exe` as the file extension.
- the archived and compressed file can be a `zip` file or `tar.gz` file
- Only one executable file is supported in the archived and compressed file
- Other files are optional, such as LICENSE
- Plugin provider should provide a sha256sum of the file that users get, it can be the sha256sum of the plugin executable file if distributed via a single file directly, or sha256sum of the archived and compressed file.

Reply:
There is a terminology section above so I added an overview in that section. I also added an example layout structure accordingly.